### PR TITLE
Combine: owner and access columns

### PR DESCRIPTION
**Pull request by shivam-sharma7**
Combine: owner and access columns AllTests.tsx
Combine: owner and access columns AllSchema.tsx
Combine: owner and access columns TestRuns.tsx
Combine: owner and access columns TestDatasets.tsx

 Combine: owner and access columns

 ## Fixes Issue: #404 #405 #428 #430

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jesperpedersen** [comment]
Still using `<AccessIcon>` -- you need an `<AccessIconOnly>` component

**Comment by shivam-sharma7** [comment]
> AccessIconOnly

where is this locate?

**Comment by jesperpedersen** [comment]
You havn't changed `<AccessIcon>`

**Comment by shivam-sharma7** [comment]
> You havn't changed `<AccessIcon>`

Okay, will do that

**Comment by jesperpedersen** [comment]
Absolutely, not.

`<AccessIcon>` is used in multiple scenarios, and most of them will require the text

**Comment by shivam-sharma7** [comment]
> Absolutely, not.
> 
> `<AccessIcon>` is used in multiple scenarios, and most of them will require the text

UI again not visible to me so you can understand what I am facing.

**Comment by shivam-sharma7** [comment]
> 

See the current changes which I made in `<AccessIcon>`

**Comment by johnaohara** [comment]
@shivam-sharma7 take a look at the ternary operator for react: https://react.dev/learn/conditional-rendering#conditional-ternary-operator--

or even the AND operator: https://react.dev/learn/conditional-rendering#logical-and-operator-

Conditional in-line rendering makes the component easier to read and eliminates duplicated code

**Comment by johnaohara** [comment]
please can you change the `[AccessIcon.tsx]` component to use in-line conditional rendering

**Comment by shivam-sharma7** [comment]
> please can you change the `[AccessIcon.tsx]` component to use in-line conditional rendering

@johnaohara Sure!

File: webapp/src/components/AccessIcon.tsx [hunk 1]
```diff
@@ -32,13 +32,16 @@ export default function AccessIcon({ access }: { access: Access }) {
             break
         }
     }
-    return (
+
+    return text ? (
```
Comment (jesperpedersen) [inline webapp/src/components/AccessIcon.tsx:36]: `text` is always set -- you need to create a new property on the component - like `notext` (default false), and use that

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: webapp/src/components/AccessIcon.tsx [hunk 2]
```diff
@@ -32,13 +32,16 @@ export default function AccessIcon({ access }: { access: Access }) {
...
         <>
             <LockedIcon style={{ fill: "var(" + color + ")" }} />
             {"\u00A0"}
             <Tooltip content={explanation}>
                 <span>{text}</span>
             </Tooltip>
         </>
+    ) : (
+        <LockedIcon style={{ fill: "var(" + color + ")" }} />
```
Comment (jesperpedersen) [inline webapp/src/components/AccessIcon.tsx:45]: You only need to guard the space and `<Tooltip>`

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: webapp/src/components/AccessIcon.tsx (hunk 1 above)
Comment (shivam-sharma7) [reply webapp/src/components/AccessIcon.tsx:36]: You mean `return notext`?

**Comment by jesperpedersen** [comment]
Stop it. We are waiting on #537 atm

**Comment by jesperpedersen** [comment]
I told you to be quiet, John and I know about the ongoing issues

**Comment by jesperpedersen** [comment]
We have jobs to do, and your patches are a priority when they are on the top of the list

**Comment by shivam-sharma7** [comment]
> I told you to be quiet, John and I know about the ongoing issues

Sorry

**Comment by shivam-sharma7** [comment]
@jesperpedersen 

image

image

**Comment by shivam-sharma7** [comment]
image

**Comment by jesperpedersen** [comment]
For these components it shouldn't say "Public"

**Comment by jesperpedersen** [comment]
yes

**Comment by jesperpedersen** [comment]
Please, reopen once done

**Pull request closed without merging**